MSAL/src/native_auth/public/MSALNativeAuthPublicClientApplication.swift

```diff
@@ -0,0 +1,312 @@
+    /// - Parameters:
+    ///   - config: Configuration for PublicClientApplication
+    ///   - challengeTypes: The set of capabilities that this application can support as an ``MSALNativeAuthChallengeTypes`` optionset
+    /// - Throws: An error that occurred creating the application object
+    public init(
+        configuration config: MSALPublicClientApplicationConfig,
+        challengeTypes: MSALNativeAuthChallengeTypes) throws {
+        guard let ciamAuthority = config.authority as? MSALCIAMAuthority else {
+            throw MSALNativeAuthInternalError.invalidAuthority
+        }
+
+        self.internalChallengeTypes =
+            MSALNativeAuthPublicClientApplication.getInternalChallengeTypes(challengeTypes)
+
+        var nativeConfiguration = try MSALNativeAuthConfiguration(
+            clientId: config.clientId,
+            authority: ciamAuthority,
+            challengeTypes: internalChallengeTypes
+        )
+        nativeConfiguration.sliceConfig = config.sliceConfig
+
+        self.controllerFactory = MSALNativeAuthControllerFactory(config: nativeConfiguration)
+        self.inputValidator = MSALNativeAuthInputValidator()
+
+        try super.init(configuration: config)
+    }
+
+    /// Initialize a MSALNativePublicClientApplication.
+    /// - Parameters:
+    ///   - clientId: The client ID of the application, this should come from the app developer portal.
+    ///   - tenantSubdomain: The subdomain of the tenant, this should come from the app developer portal.
+    ///   - challengeTypes: The set of capabilities that this application can support as an ``MSALNativeAuthChallengeTypes`` optionset
+    ///   - redirectUri: Optional. The redirect URI for the application, this should come from the app developer portal. 
+    /// - Throws: An error that occurred creating the application object
+    public init(
+        clientId: String,
+        tenantSubdomain: String,
+        challengeTypes: MSALNativeAuthChallengeTypes,
+        redirectUri: String? = nil) throws {
+        let ciamAuthority = try MSALNativeAuthAuthorityProvider()
+                .authority(rawTenant: tenantSubdomain)
+
+        self.internalChallengeTypes = MSALNativeAuthPublicClientApplication.getInternalChallengeTypes(challengeTypes)
+        let nativeConfiguration = try MSALNativeAuthConfiguration(
+            clientId: clientId,
+            authority: ciamAuthority,
+            challengeTypes: internalChallengeTypes
+        )
+
+        self.controllerFactory = MSALNativeAuthControllerFactory(config: nativeConfiguration)
+        self.inputValidator = MSALNativeAuthInputValidator()
+
+        let configuration = MSALPublicClientApplicationConfig(
+            clientId: clientId,
+            redirectUri: redirectUri,
+            authority: ciamAuthority
+        )
+
+        // we need to bypass redirect URI validation because we don't need a redirect URI for Native Auth scenarios
```

Comment:
Do you support a fallback to the web auth? If yes, we will need a redirect uri there.

Reply:
Hi Sergei, right now we don't enforce the external dev to support fallback to web auth. We still have an internal open conversation where we need to decide if we want to require to support web fallback or not.